truss/tests/test_config.py

```diff
@@ -315,29 +315,57 @@ def test_empty_config():
 
 
 def test_from_yaml():
-    yaml_path = Path("test.yaml")
     data = {"description": "this is a test"}
-    with yaml_path.open("w") as yaml_file:
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as yaml_file:
+        yaml_path = Path(yaml_file.name)
         yaml.safe_dump(data, yaml_file)
 
-    result = TrussConfig.from_yaml(yaml_path)
-
-    assert result.description == "this is a test"
+        result = TrussConfig.from_yaml(yaml_path)
 
-    yaml_path.unlink()
+        assert result.description == "this is a test"
 
 
 def test_from_yaml_empty():
-    yaml_path = Path("test.yaml")
     data = {}
-    with yaml_path.open("w") as yaml_file:
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as yaml_file:
+        yaml_path = Path(yaml_file.name)
         yaml.safe_dump(data, yaml_file)
 
-    result = TrussConfig.from_yaml(yaml_path)
+        result = TrussConfig.from_yaml(yaml_path)
+
+        # test some attributes (should be default)
+        assert result.description is None
+        assert result.spec_version == "2.0"
+        assert result.bundled_packages_dir == "packages"
+
+
+def test_from_yaml_secrets_as_list():
+    data = {"description": "this is a test", "secrets": ["foo", "bar"]}
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as yaml_file:
+        yaml_path = Path(yaml_file.name)
+        yaml.safe_dump(data, yaml_file)
+
+        with pytest.raises(ValueError):
```

Comment:
Why did we add this? If making any changes to this, the right thing to do would be making secrets a list (`Iterable[str]` actually) in the first place (and name it more appropriately "secret_names" or "secret_keys".

Reply:
this is just adding validations. right now, if you pass an iterable, it will fails server-side and does not give the user a good error.

changing this to be an iterable to begin with will break all existing trusses, so that's a much more involved change (and there are some drawbacks to it too)

Reply:
I understand. Though my suggestion was not to break all trussed, more like have a graceful handling of either-or...